manifests/install.pp

```diff
@@ -50,10 +50,11 @@
         }
 
         apt::source { 'virtualbox':
-          location => 'http://download.virtualbox.org/virtualbox/debian',
-          release  => $::lsbdistcodename,
-          repos    => $apt_repos,
-          require  => Apt::Key[ $apt_key_thumb ],
+          architecture => $::virtualbox::params::architecture,
```

Comment:
IMO it is enough to use directly facts here.
```suggestion
          architecture => $facts['os']['architecture']
```
We prefer to use structured facts instead of string facts.
And so you can let `params.pp` untouched.